_posts/2019-03-26-technical-debt.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Dług techniczny
+excerpt: Porównałem dług techniczny do długu finansowego — efektywne narzędzie do ominięcia braku zasobów w jednym momencie, licząc na to, że w późniejszym czasie będzie nam łatwiej „spłacić” takie zobowiązanie. Wspomniałem o nieoczywistych momentach, w których zaciągamy dług, jak np. podczas doboru nieodpowiedniej technologii, odchodzenie od rynkowych standardów albo... nic nie robiąc, bo dług techniczny sam z siebie narasta. Wspomniałem też o tym jak sobie z nim radzić w organizacji — przede wszystkim poprzez mierzenie jego poziomu oraz aktywną pracę z Klientem, aby utrzymywać go na komfortowym poziomie
+category: technology
+date: 2019-03-26 22:34
+author: puck
+layout: post
+---
+
+Słowem wstępu.
+
+Prowadziłem ostatnio retrospekcję jednego z projektów. Przeglądając notatki z sekcji „problemy” zaczął kształtować się jeden, bardzo jasny komunikat — mamy problem z długiem technicznym. Nie był to też odosobniony przypadek, a w dodatku objawiał się w tym projekcie w szczególny — oraz dość nieprzewidywany — sposób. Był to doskonały pretekst, żeby podsumować sobie czym jest dług techniczny, co wpływa na jego pogłębianie oraz jak nim zarządzać.
+
+## Ale dług jest techniczny czy technologiczny?
+
+Nie zbaczając za daleko z tematu: ja piszę o technicznym, jeśli ktoś życzy sobie dowiedzieć czegoś o technologicznym, to zapraszam w inne miejsce.
+
+## Czym jest dług techniczny?
+
+Tak jak każdy dług, jest to **przeniesienie pewnych zasobów w czasie** — w tym przypadku chodzi o umiejętności programistów, czas lub przychylne spojrzenie Klienta. Tworzymy oprogramowanie pożyczając sobie te rzeczy z przyszłości, licząc na to, że w późniejszej dacie będzie nam łatwiej o te rzeczy, wraz z nawiązką. 
+
+Tak jak w przypadku zadłużenia finansowego, mówimy o normalnym narzędziu, które jest bardzo popularnie używane i wcale nie należy go unikać, a jedynie **kontrolować jego poziom**.
```

Comment:
"Tak jak w przypadku zadłużenia finansowego (...) wcale nie należy go unikać" - polemizowałabym z tym stwierdzeniem. Moim zdaniem nie tyle nie należy go unikać (to dla brzmi tak, jakby warto było z niego korzystać - warto się zadłużać?), co sięgać po nie, jeżeli wymagają tego okoliczności. To moje zdanie :) 

I może "o narzędziu", a nie "o normalnym narzędziu"? Bo to jest nacechowane niepotrzebnie.

Reply:
uwzględnie. co do tego długu finansowego — spora część społeczeństwa ma kredyt hipoteczny; firmy również często lecą na linii kredytowej, dlatego IMO to jest codzienne narzędzie, a nie wyjątek.